### DISCUSSION TITLE
Using rules to set a maximum numbers of posts for an user

### DISCUSSION THREAD

**Discussion by superboss224** [pocketbase/pocketbase#3543] [Q&A]
If I have an app where I want an user to only have 10 posts on his profile, and each post has an user field that is a relation to its creator

How can I set a create rule that only lets an user create a post if he has less than 10 posts ?

**Comment by ganigeorgiev** [comment]
There are multiple ways you can approach this, but I think the easiest one would be to create a helper View collection that will keep track of the count for you and then you can join that helper collection with `@collection.*` in your original collection rule (_the only drawback of this approach is that it is prone to races, aka. in very rare cases it is possible 1 create event to slip between the check and the actual persistence, as we don't wrap the record create in a long transaction for performance reasons_).

For example:

1. Create a helper `postsCount` view collection with the following query:
    ```sql
    SELECT
        posts.user as id,
        posts.user, -- optional field to allow user expand
        count(*) as totalPosts
    FROM posts
    GROUP BY posts.user
    ```

2. Join the created view collection in your main collection API rule:
    ```js
    @request.auth.id != "" &&
    @collection.postsCount.user ?= @request.auth.id &&
    @collection.postsCount.totalPosts ?<= 10 &&
    ... other rules ...
    ```
_(note that the create API rule is executed after "dry submit", meaning that totalPosts should include the current created record)_
